### PR TITLE
Mobile category nav

### LINKED ISSUE #5417: PNI - mobile category nav (dropdown)
Implement mobile category nav (dropdown)

---
###  Design -> Dev Handoff Checklist

- [x] Design has been finalized
- [x] Link to related ticket:  #5314
- [x] Link to Abstract (make sure it can be viewed by anyone)
- [x] Design changes been merged into the master Sketch file
    - Merged into master Sketch file via Abstract. (_Master > Buyers Guide.sketch > [Home Page Updates (Q4 2020](https://share.goabstract.com/07a9aca5-3903-4f00-828a-6ac8bebb5be3?sha=e49d77cc9da6a70e8c412cd985e28138a3091cd5))_).
- [x] Desktop and mobile mockups are included
  - [[Mobile] Homepage](https://share.goabstract.com/e9fd07c9-605e-4ca9-b99c-7a3686de51cb?collectionLayerId=3f690fff-c322-4f11-8505-04aee49fea04&mode=design)
  - [[Mobile] Homepage - Secondary (Category) Navigation Expanded](https://share.goabstract.com/e9fd07c9-605e-4ca9-b99c-7a3686de51cb?collectionLayerId=c1d22a98-9658-4c0f-94b8-a44f6266ca3f&mode=design)
- [x] Name of the Sketch file (and specific page/section to look at, if applies)
- [x] Assets are included. Link: [icon accordion](https://drive.google.com/open?id=1kjM1mJFk2PfgrStEjpagg6Sk9NCn3dFO&authuser=sabrina%40mozillafoundation.org&usp=drive_fs)
- [x] Hover state for elements are designed: Match hover state on https://foundation.mozilla.org/en/who-we-are/ example mobile dropdown
PNI_secondary-nav-hover-mobile

### PR DESCRIPTION
**Pull request by mmmavis**
Repurpose foundation's site mobile 'multipage-nav' for PNI & add styling overrides to suit our needs

Closes #5417

https://foundation-s-issue-5417-3huxg6.herokuapp.com/en/privacynotincluded/

----

### 1. Dropdown items

Because mobile nav involved with complicated JS and HTML/CSS work, I wanted to reuse as much existing code as possible. When our existing mobile secondary nav expands, the nav links start after the first row (where the arrow is located). See https://foundation.mozilla.org/en/who-we-are/  example below:

Since I'm reusing most of the code with some styling changes for PNI mobile category nav, category nav links will show in a similar matter -- instead of showing links from the orange "All", we show everything after the first row. See below

This is not exactly what [the design](https://app.abstract.com/projects/fa246750-1c19-11e8-b2bf-9db5acc6ed5d/branches/master/commits/6332c2eba53b5d2906c4ec32d3ff8361276c9ad5/files/DBF82218-05B7-47F9-AD7C-8ABDFD109060/layers/18F86C85-5EF2-4DC8-A145-4D526C5540C5?collectionId=2d92eaf9-cfcb-4aa9-916d-d1a456f6e2cf&collectionLayerId=c1d22a98-9658-4c0f-94b8-a44f6266ca3f&mode=design) is for PNI but I hope we can make some compromise here to reduce code overhead. (Note that I've also reused the existing dropdown arrow icon instead of what was provided in the ticket)

### 2. Tablet category nav looks misaligned?

Because our site primary nav is already not following Bootstrap grid and same with the main PNI body content, the mobile category nav that got sandwiched in between looks kinda misaligned. Would love design feedback on this.

image

### CODE REVIEW

**Comment by sabrinang** [comment]
1. Thanks for pointing this out. I personally prefer it being at the top however I didn't really notice it was like this on the other foundation site like you've pointed out so for our quick turn around and alignment sake lets keep it to reduce code overhead. _I've raised this with designers to ask about skipping the first row in the original design and if they had thoughts._
 
2. This makes it tricky when the nav is already outside the grid to align with... but also the elements below are also off alignment. It would be ideal if we can align all the elements together _(see below left)_, but if the navigation has to protrude out from the boundary then hopefully all the other contents can align _(see below right)_. I think it would help if search was expanded like mobile style for tablet to align with easier (if there's enough space can be beside checkbox, if not have it separate etc.).
image

**Comment by mmmavis** [comment]
@sabrinang 

1. Thanks for raising this issue with the design team. I agree if we wanna update the existing dropdown we should make sure all other secondary nav dropdowns on the site are updated as well.

2. It's the same situation for primary nav - Foundation site & MozFest etc all have the same issue (i.e., protruding out of the Bootstrap boundary) so if we wanna fix this, it will have to be a separate ticket as the code is heavily re-used on all sites (Foundation/PNI/MozFest). 
I'll file a new ticket to adjust the search bar width so it's narrower and the same as the category dropdown nav width on tablet and below. This PR should be ready to review now :)

**Comment by mmmavis** [comment]
Adjusted padding within the mobile category nav.

image

**Review by sabrinang — APPROVED** [review]
Thanks for adjusting the margins and for filing issue for other elements below.

File: network-api/networkapi/buyersguide/templates/bg_base.html [hunk 1]
```diff
@@ -78,12 +78,14 @@
   </div>
 
   {% block hero %}{% endblock %}
-  <nav class="category-nav text-center" title="{% trans "site navigation" context "Tooltip on menu items" %}">
+
+  {% include "partials/multipage_nav_mobile.html" with pni_category_nav=True wrapper_classes="mt-0" %}
+  <nav id="multipage-nav" class="pni-category-nav text-center d-none d-md-block" title="{% trans "site navigation" context "Tooltip on menu items" %}">
```
Comment (KalobTaulien) [inline network-api/networkapi/buyersguide/templates/bg_base.html:83]: Nice use of `context` here!

**Review by KalobTaulien — APPROVED** [review]
Code looks good to me. Nice work @mmmavis!

**Comment by mmmavis** [comment]
Woohooo merging! 🎉

**Pull request merged**